src/java_tools/junitrunner/javatests/com/google/testing/coverage/jacoco_runner_integration_tests.sh

```diff
@@ -0,0 +1,44 @@
+#!/bin/bash
+#
+# Copyright 2012 The Bazel Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Integration testing of the JacocoCoverageRunner.
+#
+
+[ -z "$TEST_SRCDIR" ] && { echo "TEST_SRCDIR not set!" >&2; exit 1; }
+
+# Load the unit-testing framework
+source "$1" || \
+  { echo "Failed to load unit-testing framework $1" >&2; exit 1; }
+
+set +o errexit
+
+JACOCO_RUNNER="${PWD}/$2"
+shift 2
+
+JACOCO_CMD="${JACOCO_RUNNER} --main_advice=com.google.testing.coverage.JacocoCoverageRunner"
+#######################
+
+function test_jar_with_plus_in_the_filename() {
+  touch "Test.class"
```

Comment:
I'm now wondering if this is too brittle. Ideally we'd want the build below to succeed. Otherwise, if something else breaks, this test can spuriously pass. Maybe we add a simple java file in this directory, and pass the class jar in here instead of this empty .class file.

Reply:
Yes, make sense. See if the updated test is ok.